src/tree/updater_gpu_common.cuh

```diff
@@ -199,9 +199,13 @@ inline void dense2sparse_tree(RegTree* p_tree,
 
 struct BernoulliRng {
   float p;
-  int seed;
+  uint32_t seed;
+  constexpr static double conv = static_cast<double>(thrust::default_random_engine::max) /
+                                 common::GlobalRandomEngine::max();
 
-  __host__ __device__ BernoulliRng(float p, int seed) : p(p), seed(seed) {}
+  __host__ __device__ BernoulliRng(float p, size_t seed_) : p(p) {
```

Comment:
I think you don't need conv here. 
seed = static_cast<uint32_t>(seed_);
would be equivalent to
seed = seed_ % pow(2,32) ;
I think this is fine for our purposes.